src/crawler/index.ts

```diff
@@ -0,0 +1,121 @@
+import type { Nuxt } from '@nuxt/schema'
+import algoliasearch from 'algoliasearch'
+import type { SearchClient, SearchIndex } from 'algoliasearch'
+import metascraper, { Metadata } from 'metascraper'
+import type { AlgoliaOptions, CrawlerPage, GeneratePageArg } from 'src/types'
+
+/**
+ * Create a function to specify which routes should be indexed.
+ */
+function createShouldInclude (options: AlgoliaOptions) {
+  const include = options.crawler?.include ?? (() => true)
+
+  return typeof include === 'function'
+    ? include
+    : (route: string) => include.some(pattern => route.match(pattern))
+}
+
+/**
+ * Create a function to collect the routes' metadata.
+ */
+function createMetaGetter (options: AlgoliaOptions) {
+  const meta = options.crawler?.meta
+
+  if (typeof meta === 'function') {
+    return meta
+  }
+
+  const defaultMetaGetter = createDefaultMetaGetter()
+
+  if (Array.isArray(meta)) {
+    return async (html: string, route: string) => {
+      const metadata = await defaultMetaGetter(html, route)
+
+      return meta.reduce((acc, key) => ({ ...acc, [key]: metadata[key] }), {} as Metadata)
+    }
+  }
+
+  return defaultMetaGetter
+}
+
+/**
+ * Default metadata getter using "metascaper".
+ */
+function createDefaultMetaGetter () {
+  const scraper = metascraper([])
+
+  return async (html: string, route: string) => {
+    return await scraper({ html, url: route })
+  }
+}
```

Comment:
What do you think about extracting these helper functions into src/utils/index.ts? These functions are quite useful but for someone who would like to focus on the main purpose of this file they are not that important. Buy moving them to utils we are extracting the logic to other files which makes it easier to analyze :)

Reply:
Hey :) Sure, I can do that but moving it into a utils folder make it seem like they could be reused for some other purpose and in this case I don't really know what they could be used for besides that specific feature?